core/common/buffer.cpp

```diff
@@ -67,13 +66,21 @@ namespace kagome::common {
     return data_[index];
   }
 
-  Buffer Buffer::from_hex(const std::string &hex) {
-    return Buffer(std::move(unhex(hex)));
+  expected::Result<Buffer, UnhexErrors> Buffer::fromHex(std::string_view hex) {
+    return unhex(hex).match(
```

Comment:
Is there any operator which can take Result and if it contains Value, then do `map` on this Value. If error, then propagate erorr?

Like 
```
expected::Result<Buffer, UnhexErrors> Buffer::fromHex(std::string_view hex) {
    return unhex(hex).map([](const std::vector<uint8_t> &v){return Buffer(v);});
}